src/build/functions/server.ts

```diff
@@ -129,6 +133,25 @@ const writeHandlerFile = async (ctx: PluginContext) => {
   await writeFile(join(ctx.serverHandlerRootDir, `${SERVER_HANDLER_NAME}.mjs`), handler)
 }
 
+const writeRunConfig = async (ctx: PluginContext, standaloneNextConfig: NextConfigComplete) => {
+  // write our run-config.json to the root dir so that we can easily get the runtime config of the required-server-files.json
+  // without the need to know about the monorepo or distDir configuration upfront.
+  await writeFile(
+    join(ctx.serverHandlerDir, RUN_CONFIG_FILE),
+    JSON.stringify({
+      nextConfig: standaloneNextConfig,
+      // only enable setting up 'use cache' handler when Next.js supports CacheHandlerV2 as we don't have V1 compatible implementation
+      // see https://github.com/vercel/next.js/pull/76687 first released in v15.3.0-canary.13
+      enableUseCacheHandler: ctx.nextVersion
+        ? satisfies(ctx.nextVersion, '>=15.3.0-canary.13', {
+            includePrerelease: true,
+          })
+        : false,
+    } satisfies RunConfig),
+    'utf-8',
+  )
+}
```

Comment:
this is to make `run-config.json` file to also contain next-runtime toggles and not just Next.js config to allow us to conditionally enable some setups - in this case conditionally setup our `use cache` handler depending on version because using implementation in this PR that does not implement V1 variant is resulting in crashes due to some methods from V1 interface not being implemented

Reply:
This was moved since initially commented on, but leaving comment up for visibility as same change is still in this PR, just in slightly different place (to reduce amount of changes it introduces)